### PR TITLE
add PodDisruptionBudget to otel-operator

### PR DESCRIPTION
**Pull request by avoidik**
hello,

with this PR I'd like to introduce PodDisruptionBudget configuration to otel-operator itself.

### CODE REVIEW

**Comment by Allex1** [comment]
@avoidik thanks for your contribution. Pls bump the chart version and run `make generate-examples`

File: charts/opentelemetry-operator/templates/pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  minAvailable: {{ .Values.pdb.minAvailable }}
+  {{- end }}
+  {{- if .Values.pdb.maxUnavailable }}
+  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "opentelemetry-operator.selectorLabels" . | nindent 6 }}
+      app.kubernetes.io/component: controller-manager
+{{- end }}
```
Comment (Allex1) [inline charts/opentelemetry-operator/templates/pdb.yaml:21]: nl

File: charts/opentelemetry-operator/examples/default/rendered/deployment.yaml [hunk 2]
```diff
@@ -22,10 +22,10 @@ spec:
     metadata:
       annotations:
         kubectl.kubernetes.io/default-container: manager
-        
+        aaa: bbb
```
Comment (avoidik) [inline charts/opentelemetry-operator/examples/default/rendered/deployment.yaml:25]: here `nindent` render spaces if `podAnnotations` is empty

File: charts/opentelemetry-operator/examples/default/rendered/deployment.yaml [hunk 3]
```diff
@@ -22,10 +22,10 @@ spec:
...
       labels:
         app.kubernetes.io/name: opentelemetry-operator
-        
+        ccc: ddd
```
Comment (avoidik) [inline charts/opentelemetry-operator/examples/default/rendered/deployment.yaml:28]: here `nindent` render spaces if `podLabels` is empty

File: charts/opentelemetry-operator/examples/default/values.yaml [hunk 4]
```diff
@@ -1 +1,7 @@
-
+pdb:
+  create: true
```
Comment (avoidik) [inline charts/opentelemetry-operator/examples/default/values.yaml:2]: opt-in

Comment (avoidik) [inline charts/opentelemetry-operator/examples/default/values.yaml:2]: opt-in

File: charts/opentelemetry-operator/values.yaml [hunk 5]
```diff
@@ -11,6 +11,19 @@ nameOverride: ""
 ## Reference one or more secrets to be used when pulling images from authenticated repositories.
 imagePullSecrets: []
 
+## Pod Disruption Budget configuration
+##
+pdb:
+  ## Enable/disable a Pod Disruption Budget creation
+  ##
+  create: false
```
Comment (avoidik) [inline charts/opentelemetry-operator/values.yaml:19]: opt-out by default

File: charts/opentelemetry-operator/examples/default/values.yaml (hunk 4 above)
Comment (Allex1) [reply charts/opentelemetry-operator/examples/default/values.yaml:2]: I don;t think this is needed

File: charts/opentelemetry-operator/examples/default/rendered/deployment.yaml (hunk 2 above)
Comment (Allex1) [reply charts/opentelemetry-operator/examples/default/rendered/deployment.yaml:25]: you can fix it in  https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/templates/deployment.yaml#L19

File: charts/opentelemetry-operator/Chart.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: opentelemetry-operator
-version: 0.24.0
+version: 0.25.0
```
Comment (Allex1) [inline charts/opentelemetry-operator/Chart.yaml:3]: This needs to be 0.24.1 (and run `make generate-examples` again) . We should also update the [docs](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/CONTRIBUTING.md) to specify patch version and not minor.

File: Makefile [hunk 7]
```diff
@@ -28,15 +28,15 @@ generate-examples:
...
+		EXAMPLES=$$(find $${EXAMPLES_DIR} -maxdepth 1 -mindepth 1 -type d -exec basename \{\} \;); \
 		for example in $${EXAMPLES}; do \
 			echo "Checking example: $${example}"; \
 			VALUES=$$(find $${EXAMPLES_DIR}/$${example} -name *values.yaml); \
 			for value in $${VALUES}; do \
 				helm dependency build charts/$${chart_name}; \
 				helm template example charts/$${chart_name} --namespace default --values $${value} --output-dir "${TMP_DIRECTORY}/$${example}"; \
 				SUBCHARTS_DIR=${TMP_DIRECTORY}/$${example}/$${chart_name}/charts; \
-				SUBCHARTS=$$(find $${SUBCHARTS_DIR} -type d -maxdepth 1 -mindepth 1 -exec basename \{\} \;); \
+				SUBCHARTS=$$(find $${SUBCHARTS_DIR} -maxdepth 1 -mindepth 1 -type d -exec basename \{\} \;); \
```
Comment (dmitryax) [inline Makefile:39]: Why were these changes added? Let's avoid unrelated diff parts

File: charts/opentelemetry-operator/templates/deployment.yaml [hunk 8]
```diff
@@ -16,10 +16,14 @@ spec:
     metadata:
       annotations:
         kubectl.kubernetes.io/default-container: manager
+        {{- if .Values.manager.podAnnotations }}
```
Comment (avoidik) [inline charts/opentelemetry-operator/templates/deployment.yaml:19]: just to be able to evade `nindent` if properties are blank

File: charts/opentelemetry-operator/templates/deployment.yaml [hunk 9]
```diff
@@ -16,10 +16,14 @@ spec:
...
         {{- include "opentelemetry-operator.podAnnotations" . | nindent 8 }}
+        {{- end }}
       labels:
         {{- include "opentelemetry-operator.selectorLabels" . | nindent 8 }}
+        {{- if .Values.manager.podLabels }}
```
Comment (avoidik) [inline charts/opentelemetry-operator/templates/deployment.yaml:24]: just to be able to evade `nindent` if properties are blank

Comment (avoidik) [reply charts/opentelemetry-operator/templates/deployment.yaml:24]: as alternative to conditions I could try to move `nindent` into the helper method(s) instead, let me know if I should do something

Comment (Allex1) [reply charts/opentelemetry-operator/templates/deployment.yaml:24]: this works

**Review by Allex1 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]
